### DISCUSSION TITLE
「今日のトレンド記事」と「関連記事」の表示順序　および　「関連記事」の表示の確実化　について

### DISCUSSION THREAD

**Discussion by tsubolabo** [increments/qiita-discussions#811] [要望・アイデア・不具合]
最近、各記事の最後に「今日のトレンド記事」が表示されるようになりました。以前からあった「関連記事」と一緒に表示されます。しかし、まだ試行中のためか？　両者の表示順序が上になったり、下になったり、その都度変わっているようにみえます。私としては、「関連記事」が先に表示される方が便利です。

　今までも、「関連記事」が有益な情報に誘導してくれたことがたびたびありました。一方、「今日のトレンド記事」には、興味を持って閲覧中の記事の内容とは殆ど関係ないものがリストアップされるだけなので、あまりクリックすることはありません。「今日のトレンド記事」が不要とまでは言いませんが、この順序は後の方が有難いと思います。

　ところで、「関連記事」ですが、本来であれば図２のような表示になるべきところ、図１のような表示のままで、いつまで待っても記事リストが表示されない不具合がたびたび発生します。以前にも#429（都合により弊文は削除）で改善を要望していますが、実現していません。「関連記事」は役立つリストなので、早期の改善をお願いします。

私のパソコンの OS は Windows 11, version 23H2。

今日のところは、図１が Microsoft Edge バージョン 125.0.2535.67 (公式ビルド) (64 ビット)、図２が Google Chrome バージョン 125.0.6422.78 (Official Build) (64 ビット)による表示ですが、どちらのブラウザでも、そのときどきで図１の状態が発生したりしなかったりします。

**図１　　関連記事が正しく表示されない画面**
<img width="652" alt="image01" src="https://github.com/increments/qiita-discussions/assets/42261605/506a3bb2-5080-4606-88c2-c7ae10f39873">

**図２　　関連記事の正しい表示画面**
<img width="649" alt="image02" src="https://github.com/increments/qiita-discussions/assets/42261605/121cff81-7024-41d4-b03d-d06a946f6ca9">

**Comment by tanaka-asami415** [comment]
@tsubolabo 
> 各記事の最後に「今日のトレンド記事」が表示されるようになりました。以前からあった「関連記事」と一緒に表示されます。しかし、まだ試行中のためか？　両者の表示順序が上になったり、下になったり、その都度変わっているようにみえます。

現在、「関連記事」の表示位置のテストを行っており、5/27に場所の入れ替えを行いました。
また、今後も様々なテストを行う予定があり、同様の現象が発生する場合がございます。ご了承いただけると幸いです🙇

「関連記事」の記事リストが表示されない件については、現在も原因の特定には至っておらず申し訳ありません。
以前、AdBlock等の拡張機能が影響している可能性もあると他の方よりコメントを頂いておりましたが
拡張機能のオフを試していただいている状態でも、症状が出ているでしょうか？

**Reply by tsubolabo** [reply]
@tanaka-asami415 

> 現在、「関連記事」の表示位置のテストを行っており、5/27に場所の入れ替えを行いました。また、今後も様々なテストを行う予定があり、同様の現象が発生する場合がございます。

　了解しました。「関連記事」が上位置で定着することを期待しています。

> 「関連記事」の記事リストが表示されない件については、現在も原因の特定には至っておらず申し訳ありません。以前、AdBlock等の拡張機能が影響している可能性もあると他の方よりコメントを頂いておりましたが拡張機能のオフを試していただいている状態でも、症状が出ているでしょうか？

　AdBlock はしていないので、その可能性はないと独自判断して対応はしませんでした。

　現在、Chrome の拡張機能には「Adobe Acrobat」「F-Secureのブラウザ保護」「Markdown Viewer」、Edge の拡張機能には「F-Secureのブラウザ保護」「Markdown Viewer」が入っています。「Markdown Viewer」は意図して入れたものですが、その他は何となく入ってきたものです。

　「F-Secureのブラウザ保護」は、Niftyのセキュリティソフト「常時安全セキュリティ24」（バージョン19.2）のインストール時に入ったものと思われます。少々怪しそうではありますが、セキュリティソフトは、種類が異なるにしても誰でも使っているものだろうし、なくすのも脅威がありそうなので、オフにしたことはありません。

**Reply by tanaka-asami415** [reply]
@tsubolabo 
> AdBlock はしていないので、その可能性はないと独自判断して対応はしませんでした。

そうなんですね。
拡張機能については、個別にオフにしていただかなくても、シークレットモード等で確認していただくことが可能です。

AdBlock以外の拡張機能が影響している可能性も0ではないため、念のため一度お試しいただけますと幸いです。
拡張機能なしの状態で試す方法としては下記の方法があります。

- Microsoft Edge「[InPrivateブラウザ](https://support.microsoft.com/ja-jp/microsoft-edge/microsoft-edge-%E3%81%A7-inprivate-%E3%83%96%E3%83%A9%E3%82%A6%E3%82%BA%E3%82%92%E4%BD%BF%E3%81%86-e6f47704-340c-7d4f-b00d-d0cf35aa1fcc)」
- Chrome「[シークレットモード](https://support.google.com/chrome/answer/95464?hl=ja&co=GENIE.Platform%3DDesktop)」

よろしくお願いします。

**Reply by tsubolabo** [reply]
@tanaka-asami415 

> AdBlock以外の拡張機能が影響している可能性も0ではないため、念のため一度お試しいただけますと幸いです。拡張機能なしの状態で試す方法としては下記の方法があります。

　今のところ、Edge では「新しい InPrivate ウィンドウ」でも再現しました。しかし、Chrome の「新しいシークレット ウィンドウ」ではまだ再現していません。100発100中で現れる現象ではないので、ここ何日か、新規記事のチェックを Chrome の「新しいシークレット ウィンドウ」で続けてみます。

**Reply by tsubolabo** [reply]
@tanaka-asami415 

Chrome の「新しいシークレット ウィンドウ」でも再現しました。
再現との関連は分かりませんが、念のため、ここに至った手順を書いておきます。

①　デスクトップのアイコンをクリックして Chrome を開く。
②　画面右上の縦三点リーダーから「新しいシークレット ウィンドウ」をクリック。
③　①で開いた Chrome は閉じて、以降はシークレット ウィンドウだけで操作する。
④　自分の Qiita ホーム画面の URL を入力する（以下、ログインしないままでの操作）。
⑤　ホーム画面上で「*」を検索する。
⑥　「関連順」を「新着順」に変更する。
⑦　上から順番に、興味あるタイトルだけ開いて閲覧後、「関連記事」の表示状態をチェックする。

最初のうちは「関連記事」の表示は正常でしたが、⑦を何回か繰り返しているうちに異常な表示が再現しました。一旦異常になると、以前には正常表示されていた記事でも異常表示しかされなくなります。

**Reply by tanaka-asami415** [reply]
@tsubolabo 
お試しいただいた詳細を教えていただき、ありがとうございます。
シークレットウィンドウでも再現するのですね…

改めて我々でも調査は行う予定ですが、現時点では再現することができず
ご報告いただいている事例が本件だけということもあり、原因の特定が難しい可能性もございます。予めご了承ください🙇

---

もし他のユーザーさまで、同様の現象の方がいましたら、こちらのDiscussionsへご連絡お待ちしております！

**Comment by wany-oh** [comment]
「関連記事」に記事とは全く関係ない広告が真の関連記事と全く同じフォーマットで表示されるのも問題があると思います。

**Reply by tsubolabo** [reply]
@wany-oh

　コメントをありがとうございます。そうかもしれません。

　ブラウザの更新ボタンをクリックすると、関連記事の上から1～4行目の「真の関連記事」は、ほぼ毎回 同一のタイトルが表示されます。しかし、5,6行目にあるPRのタイトルはその都度内容が変わります。

　特定のPRタイトルが含まれたときだけ、セキュリティソフトが何等かを誤感知して表示を阻止していると考えると辻褄は合います（また、「妄想や邪推による曲解」と言って叱られるかもしれませんが・・・）。

　この仮説を裏付けるため、同一記事を開いたまま、更新ボタンをクリックし続けたところ、突然 異常表示に変わりました。その状態からさらにクリックし続けたところ、また正常表示に戻りました。しかし、その後は何回クリックしても、この現象は現れていません。たまたまの現象だったのか？　試行回数が足りないだけなのか？

**Reply by tanaka-asami415** [reply]
@wany-oh 
ありがとうございます！
原因が特定できていないため、何とも言えないですが調査の際に参考にさせていただきます。

**Comment by tanaka-asami415** [comment]
@tsubolabo 
お手数をおかけするのですが、もし可能でしたら次回発生時に
コンソールでのエラー状況などをご提供いただけないでしょうか🙇

各コンソールの確認の仕方は下記をご参照ください。

- [Microsoft Edge](https://learn.microsoft.com/ja-jp/microsoft-edge/devtools-guide-chromium/console/#log-information-to-display-in-the-console:~:text=%E8%AD%A6%E5%91%8A%E3%81%97%E3%81%BE%E3%81%99%E3%80%82-,%E3%82%B3%E3%83%B3%E3%82%BD%E3%83%BC%E3%83%AB%E3%82%92%E9%96%8B%E3%81%8F%E3%81%AB%E3%81%AF%E3%80%81Ctrl%20%2B%20Shift%20%2B%20J%20(Windows%E3%80%81Linux)%20%E3%81%BE%E3%81%9F%E3%81%AF%20Command%20%2B%20Option%20%2B%20J%20(macOS)%20%E3%82%92%E6%8A%BC%E3%81%97%E3%81%BE%E3%81%99%E3%80%82,-%E3%82%B3%E3%83%B3%E3%82%BD%E3%83%BC%E3%83%AB%E3%81%AB%E3%81%AF)
- [Chrome](https://developer.chrome.com/docs/devtools/console/reference?hl=ja#open)

よろしくお願いします。

**Reply by tsubolabo** [reply]
@tanaka-asami415 

　Chrome で新規投稿を閲覧中に突然 表示がおかしくなりました。この状態のまま、「３点メニュー」>「その他のツール」>「デベロッパーツール」でコンソールを開いたときの画面を添付します。

<img width="960" alt="コンソールのエラー表示01" src="https://github.com/increments/qiita-discussions/assets/42261605/70d1cd8b-c001-4327-a2ba-12428666085a">

**Reply by tanaka-asami415** [reply]
@tsubolabo 
早速ありがとうございます！

いただいた画像をレコメンドエンジンの提供元に確認いただこうと思います。
また別途ご依頼をする事があるかもしれませんが、その際はご協力いただけると幸いです🙇

**Reply by tanaka-asami415** [reply]
@tsubolabo 
お待たせしました。
レコメンドエンジン提供元に画像を確認いただいたところ「ERR_NAME_NOT_RESOLVEDエラー」が発生しているそうです。

こちらのエラーは、ユーザー様の通信環境や端末・ブラウザに依存しているとのことでした。
対応策としては、DNSのキャッシュクリア等をしていただき、しばらく様子を見ていただけますでしょうか。

参考例として教えていただいたリンク先が分かりやすそうなため、ChromeやEdgeの公式ドキュメントではないのですが、そのままご紹介させていただきます🙏
https://kinsta.com/jp/knowledgebase/err_name_not_resolved/

よろしくお願いいたします。

**Reply by tsubolabo** [reply]
@tanaka-asami415 

> こちらのエラーは、ユーザー様の通信環境や端末・ブラウザに依存しているとのことでした。対応策としては、DNSのキャッシュクリア等をしていただき、しばらく様子を見ていただけますでしょうか。

　昨日、下記により Chrome のキャッシュを削除しました。

「３点メニュー」>「設定」>「プライバシーとセキュリティ」>「閲覧履歴データの削除」>「詳細設定」>　ここで「キャッシュされた画像とファイル」のみを選択 >　「データを削除」

　その結果、当日中の確認では、「関連記事」の表示不具合は再現せず、うまく解決したかに見えました。しかし、翌日（今朝）、また再発しました。そのときのコンソール画像を添付します。

　参考までに、このときのキャッシュフォルダー（C:\ユーザー\\(ユーザー名)\AppData\Local\Google\Chrome\UserData\Default\Cache\Cache_Data）内のファイル数は 596、容量103MB、ファイルの日付は 2024/06/05 17:27 ～ 2024/06/06 7:48 です。

<img width="960" alt="コンソールのエラー表示02" src="https://github.com/increments/qiita-discussions/assets/42261605/871ca04b-24b4-4f2f-847d-4b874b259630">

**Reply by tanaka-asami415** [reply]
@tsubolabo 
ご確認ありがとうございます。
恐らく上記で対応いただいた内容は、`ブラウザ`のキャッシュクリアかと存じます。

今回対応いただく必要があるのは、`DNS`のキャッシュクリア等となり、また別の内容になります。

https://kinsta.com/jp/knowledgebase/err_name_not_resolved/#err_name_not_resolved6
を参考に、順番に対応いただけますでしょうか🙏

よろしくお願いします🙇

**Reply by tsubolabo** [reply]
@tanaka-asami415 

　下記の結果から、Wi-Fi ルーターの劣化のようです。そろそろ買い替え時期かと思われます。個人的な問題で大変お騒がせいたしました。申し訳ありません🙇

①　ルーターの再起動（電源再投入）
　　　頻度は減ったように思えるが不具合の発生あり。
　　　表示が正常でも、ERR_NAME_NOT_RESOLVED が出ることあり。
②　DNSキャッシュのクリア（ipconfig /flushdns）
　　　①の結果と同程度。
③　ルーターへの有線接続
　　　①の結果と同程度。
④　ONUへの有線接続
　　　確認期間は十分ではないが、不具合の発生は無さそう。
　　　今のところ、ERR_NAME_NOT_RESOLVED が出ていない。

　しかし、なぜ「関連記事」だけに皺寄せがくるのか？、知りたいところです。Qiita本体用と関連記事用のサーバーの能力の差でしょうか？

**Reply by tanaka-asami415** [reply]
@tsubolabo 
お試しいただきありがとうございます！
とんでもないです。現時点での再現がなさそうとのこと、よかったです！🙌✨

> 　しかし、なぜ「関連記事」だけに皺寄せがくるのか？、知りたいところです。Qiita本体用と関連記事用のサーバーの能力の差でしょうか？

外部のレコメンドエンジンを利用しているので、詳細は分からないのですが、Qiitaとは違うサービスという点は理由の一つかもしれません。

**Reply by tsubolabo** [reply]
①
　「関連記事」の表示不具合は、当方で対策した結果、その後は発生しておらず快適に閲覧できています。ご指導ありがとうございました。ただし、対策にはかなり苦労しましたので、その顛末を記事にして投稿しました。恥の上塗りにはなりますがご参考まで。[【Wi-Fi ルータ】壊れたと思って買い換えたのに、実は壊れてはいませんでした](https://qiita.com/tsubolabo/items/28f564f12a13bb181231)

②
　以前の「今日のトレンド記事」は、現在は「[タグ名]の先週いいねが多かった記事」に変わっており、表示の順番もまた入れ替わっています（順番の入れ替わりは今後もあり得ることを聞いていましたので、特に言うことはありません）。記事の選択範囲は以前よりも絞り込まれているので、漠然としていたトレンド記事よりも、目を引く項目が多くなりました。
　当方、「関連記事」との置き換えを模索しているのでは？と「邪推」したりしているのですが、「関連記事」に比べると、あと一歩及びません。タグ名を複数にしたり、本文中の単語も検索対象にして、もう少し絞り込んでいただけると助かります。

**Reply by tanaka-asami415** [reply]
@tsubolabo 
ご報告ありがとうございます。

その後、快適に閲覧ができているとのこと何よりです✨
今後、同様の現象が起きた方に、こちらのDiscussions等をご案内することができるので、こちらこそ、お問合せや詳細を共有いただきありがとうございました。

「[タグ名]の先週いいねが多かった記事」についてのご意見もありがとうございます。
精度を高めていきたいと考えておりますので、参考にさせていただきます🙇
